pilot/pkg/networking/core/v1alpha3/networkfilter.go

```diff
@@ -167,7 +174,9 @@ func buildOutboundNetworkFilters(node *model.Proxy,
 			statPrefix = util.BuildStatPrefix(push.Mesh.OutboundClusterStatName, routes[0].Destination.Host,
 				routes[0].Destination.Subset, port, &service.Attributes)
 		}
-		return buildOutboundNetworkFiltersWithSingleDestination(push, node, statPrefix, clusterName, port)
+		destRule := push.DestinationRule(node, service)
+		destinationRule := CastDestinationRule(destRule)
+		return buildOutboundNetworkFiltersWithSingleDestination(push, node, statPrefix, clusterName, port, destinationRule)
 	}
 	return buildOutboundNetworkFiltersWithWeightedClusters(node, routes, push, port, configMeta)
```

Comment:
Doesn't it need src ip hash?

Reply:
I do not know if it makes sense for weighted clusters - Does it?

Reply:
Don't we support it for HTTP? We should be consistent at least I think

Reply:
I just looked at envoy code and it picks the cluster based on weights first applies the hash - so we should be able to do it here as well. Let me try